src/app/portlets/content-types/form/content-types-form.component.spec.ts

```diff
@@ -493,44 +494,64 @@ describe('ContentTypesFormComponent', () => {
         let data = null;
         spyOn(comp, 'submitForm').and.callThrough();
 
-        comp.submit.subscribe((res) => (data = res));
+        comp.submit.subscribe(res => (data = res));
         comp.submitForm();
 
         expect(comp.submitForm).toHaveBeenCalled();
         expect(data).toBeNull();
     });
 
-    it('should send data with valid form', () => {
-        spyOn(dotLicenseService, 'isEnterpriseLicense').and.returnValue(Observable.of(true));
-
-        comp.data = {
-            baseType: 'CONTENT'
-        };
-        fixture.detectChanges();
-
-        let data = null;
-        spyOn(comp, 'submitForm').and.callThrough();
-
-        comp.submit.subscribe((res) => (data = res));
+    describe('send data with valid form', () => {
+        let data;
 
-        comp.form.controls.name.setValue('A content type name');
-
-        fixture.detectChanges();
-        comp.submitForm();
+        beforeEach(() => {
+            spyOn(dotLicenseService, 'isEnterpriseLicense').and.returnValue(Observable.of(true));
+            comp.data = {
+                baseType: 'CONTENT'
+            };
+            fixture.detectChanges();
+            data = null;
+            spyOn(comp, 'submitForm').and.callThrough();
+            comp.submit.subscribe(res => (data = res));
+            comp.form.controls.name.setValue('A content type name');
+            fixture.detectChanges();
+        });
 
-        expect(comp.submitForm).toHaveBeenCalledTimes(1);
-        expect(data).toEqual({
-            clazz: '',
-            description: '',
-            detailPage: '',
-            host: '',
-            name: 'A content type name',
-            urlMapPattern: '',
-            defaultType: null,
-            fixed: null,
-            folder: null,
-            system: null,
-            workflow: ['85c1515c-c4f3-463c-bac2-860b8fcacc34']
+        it('should submit form correctly', () => {
+            comp.submitForm();
+
+            expect(comp.submitForm).toHaveBeenCalledTimes(1);
+            expect(data).toEqual({
+                clazz: '',
+                description: '',
+                detailPage: '',
+                host: '',
+                name: 'A content type name',
+                urlMapPattern: '',
+                defaultType: null,
+                fixed: null,
+                folder: null,
+                system: null,
+                workflow: ['85c1515c-c4f3-463c-bac2-860b8fcacc34']
+            });
+        });
+        it('should submit form correctly on Enter', () => {
+            const form = fixture.debugElement.query(By.css('form'));
+            form.nativeElement.dispatchEvent(new KeyboardEvent('keyup', { key: 'Enter' }));
+            expect(comp.submitForm).toHaveBeenCalledTimes(1);
+            expect(data).toEqual({
```

Comment:
I think this expect is not neccessary here because this is being tested in the previous test, but is ok.